### PR TITLE
Add Hashing intro

### PR DESCRIPTION
**Pull request by samuel-lucas6**
I noticed this page was blank and thought I'd chip in. It might need a little rewording.

### CODE REVIEW

**Comment by samuel-lucas6** [comment]
I've also added a reference to the [Partitioning Oracle Attacks](https://eprint.iacr.org/2020/1491.pdf) paper for the Robustness section of the AEAD page. I didn't mean to make that part of the same pull request though, my bad.

File: hashing/README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
 # Hashing
 
-(this documentation is a work in progress. Feel free to contribute a nice intro to hash functions!)
+*Hashing* refers to using a function that takes an input of any size and produces a fixed size output. Hashing the same input always produces the same output. This output is commonly known as a hash or digest.
+
+Hashes are often used to authenticate data. For example, to detect whether a file has been modified or corrupted.
```
Comment (emilbayes) [inline hashing/README.md:5]: Wouldn't "integrity" be a better word here? "authenticate" has a pretty specific meaning (implies a secret key in my mind)

Comment (jedisct1) [reply hashing/README.md:5]: Nod. A hash function doesn't authenticate a message, unless it is keyed or used in a construction such as HMAC.

The first part is not necessarily correct either. The output is deterministic for a given set of parameters (which can include the output length), not just a message.

Comment (samuel-lucas6) [reply hashing/README.md:5]: Yeah, I'll do some rewording.

Comment (samuel-lucas6) [reply hashing/README.md]: > Hashing the same input with the same function and output length always produces the same output.

Thoughts on this? Alternatively, that sentence could just be exactly what you said: 'The output is deterministic for a given set of parameters (which can include the output length), not just a message.' I was just trying to keep it simple.

**Review by samuel-lucas6 — COMMENTED** [review]
The Partitioning Oracle Attacks reference is missing a full stop after the authors in brackets compared to the first reference. I don't think a full stop is necessary there, but it's up to you.

File: hashing/README.md (hunk 1 above)
Comment (samuel-lucas6) [reply hashing/README.md]: > Hashing the same input with the same function and parameters (e.g. output length) always produces the same output.

Hopefully that's preferable.

**Comment by samuel-lucas6** [comment]
@jedisct1 I didn't mean to offend or annoy you if that's what I've managed to do. I'm just trying to improve the documentation, which is something you have listed on the [Roadmap](https://doc.libsodium.org/roadmap). Preventing people from opening issues likely limits your ability to make improvements. I guess I won't bother trying to contribute anymore since your response suggests that you would prefer me not to.

**Comment by jedisct1** [comment]
Don't take it personally. Here is why: https://00f.net/2021/03/26/it-doesnt-work/

**Comment by samuel-lucas6** [comment]
> 
> 
> Don't take it personally. Here is why: https://00f.net/2021/03/26/it-doesnt-work/

@jedisct1 I'm very sorry for contributing to the problem, and thank you for the explanation. You've done a lot of amazing work, you're a trustworthy source of information, and you explain things really well. The industry needs more people like you. A huge thank you for all of your work on FOSS projects. I hope things improve for you.

**Pull request closed without merging**